community/contributing/pr_workflow.rst

```diff
@@ -204,6 +204,20 @@ Therefore, the command that you should (almost) always use is:
 
     $ git pull --rebase upstream master
 
+
+If you have already pushed the merge commit without using ``rebase`` (or 
+have made any mistakes that have resulted in undesired history), you may
+use ``reset`` to revert to a specific commit and try again:
```

Comment:
Suggestion: you may use a hard reset. Reset itself doesn't change the history. Like @Calinou suggested, it'd be nice to explain what a hard reset is and what it does as it's a destructive operation.